README.md

```diff
@@ -66,6 +68,51 @@ where:
 * 60 * 1000 - write interval in milliseconds (optional parameter, by default interval is equal to 10 seconds)
 * 'custom-docs-path' - Swagger UI path for your REST API (default: api-docs)
 
+## Advanced usage (recommended)
+
+Instead of using a single `init` handler, we'll use 2 separate ones - one for **responses**, and one for **requests**.
+
+```diff
+let app = express();
+
+-expressOasGenerator.init(app, {});
++expressOasGenerator.injectResponseMiddleware(app, {});
+
+/** do other stuff with `app` */
+
+app.listen(PORT, () => {
++	expressOasGenerator.injectRequestMiddleware(app);
```

Comment:
I'm pretty sure we don't have to put the injectRequestMiddleware inside the app.listen() callback - we should be fine putting it right before calling app.listen, so long as it is the last middleware, right?

Because someone might have a need to export the app and have a separate function for starting the app/server (like I do in turbo-schedule). It's not a problem for me, currently, because I don't really use the exported app. But it might be useful to apply request handling middleware before exporting the app and before starting the server.

For example, testing with supertest. What I do is just start the server initially before the tests run, and then just use the localhost:{port}/{api-endpoint} string for supertest.
But someone might want to load the app into supertest everytime they make a request, and this is where a difference would matter.

Reply:
sounds good